Justfile

```diff
@@ -0,0 +1,84 @@
+# them at earliest convenience :)
+# Also in addition to this, the `nextest-integration` test is currently disabled on macos
+with_macos_excludes := if os() == "macos" {
+    "--exclude node-runtime --exclude runtime-params-estimator --exclude near-network --exclude estimator-warehouse"
+} else {
+    ""
+}
+nightly_flags := "--features nightly,test_features"
+bash := if os() == "macos" { "/bin/bash" } else { "/usr/bin/bash" }
+
+# all the tests, as close to CI as possible
+test: test-ci test-extra
+
+# only the tests that are exactly the same as the ones in CI
+test-ci: (nextest "stable") (nextest "nightly") check-style
+
+# tests that are as close to CI as possible, but not exactly the same code
+test-extra: check-lychee
+
+# all cargo tests, TYPE is "stable" or "nightly"
+nextest TYPE: (nextest-unit TYPE) (nextest-integration TYPE)
+
+# cargo unit tests, TYPE is "stable" or "nightly"
+nextest-unit TYPE:
+    cargo nextest run \
+        --locked \
+        --workspace \
+        --exclude integration-tests \
+        --cargo-profile dev-release \
+        --profile ci \
+        {{ with_macos_excludes }} \
+        {{ if TYPE == "nightly" { nightly_flags } \
+           else if TYPE == "stable" { "" } \
+           else { error("TYPE is neighter 'nightly' nor 'stable'") } }}
+
+# cargo integration tests, TYPE is "stable" or "nightly"
+[linux]
+nextest-integration TYPE:
+    cargo nextest run \
+        --locked \
+        --package integration-tests \
+        --cargo-profile dev-release \
+        --profile ci \
+        {{ if TYPE == "nightly" { nightly_flags } \
+           else if TYPE == "stable" { "" } \
+           else { error("TYPE is neither 'nightly' nor 'stable'") } }}
+[macos]
+nextest-integration TYPE:
+    @echo "Nextest integration tests are currently disabled on macos!"
+
+# generate a codecov report for RULE
+codecov RULE:
+    #!{{ bash }}
+    set -euxo pipefail
+    source <(cargo llvm-cov show-env --export-prefix)
+    just {{ RULE }}
+    cargo llvm-cov report --profile dev-release --codecov --output-path codecov.json
+
+# style checks
+check-style:
+    python3 scripts/check_nightly.py
+    python3 scripts/check_pytests.py
+    python3 scripts/fix_nightly_feature_flags.py
+    ./scripts/formatting --check
+
+# lychee-based url validity checks
+check-lychee:
+    lychee . {{ if env("GITHUB_TOKEN", "") != "" { "" } else { "-a 429" } }}
+    @echo {{ if env("GITHUB_TOKEN", "") != "" { "" } \
+             else { "Note: 'Too Many Requests' errors are allowed here but not in CI, set GITHUB_TOKEN to check them" } }}
+
+# build target/rpc_errors_schema.json
+build-rpc-errors-schema:
+    rm -f target/rpc_errors_schema.json
+    cargo check -p near-jsonrpc --features dump_errors_schema
+
+# update chain/jsonrpc/res/rpc_errors_schema.json
+update-rpc-errors-schema: build-rpc-errors-schema
+    cp target/rpc_errors_schema.json chain/jsonrpc/res/rpc_errors_schema.json
```

Comment:
Any reason these have been split up into two?

Reply:
This way local development can use `update` and CI `check`, without duplicating the code in`build` :)

Reply:
I don’t personally find (check not updating the file that is being checked) a useful thing to do, but alright.

Actually, I feel that it would still be better to check with `git diff` as it was done before, as ultimately what we care about is “is chain/jsonrpc/res/rpc_errors_schema.json the same as the checked-in file” and not “is the newly built file equal to whatever file is currently checked out (including any modifications not checked in)”.

But no strong opinion here.

Reply:
Hmm I see your point, I guess my thinking is mostly "`just test` should not affect the worktree outside of `target`, as that’d be quite unexpected" ^^’

That said, as you say, I think we can reasonably live with both options :)